src/Provider/MapQuest/GeocodeQuery.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Geocoder package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @license    MIT License
+ */
+
+namespace Geocoder\Provider\MapQuest;
+
+use Geocoder\Formatter\StringFormatter;
+use Geocoder\Location;
+use Geocoder\Query\GeocodeQuery as BaseGeocodeQuery;
+
+class GeocodeQuery extends BaseGeocodeQuery implements GetAddressInterface
+{
+    const DATA_KEY_ADDRESS = 'address';
+
+    protected function __construct($address)
+    {
+        if ($address instanceof Location) {
+            $this->data[static::DATA_KEY_ADDRESS] = $address;
+        } else {
+            parent::__construct($address);
+        }
+    }
+
+    public static function createFromAddress(Location $address): self
+    {
+        return new self($address);
+    }
+
+    public function getAddress()
+    {
+        return $this->getData(static::DATA_KEY_ADDRESS);
+    }
+
+    public function getText(): string
+    {
+        if (!$this->text) {
+            $address = $this->getAddress();
+            if ($address instanceof Location) {
+                $this->text = $this->formatAddress($address);
+            }
+        }
+
+        return $this->text;
+    }
+
+    protected function formatAddress(Location $address): string
```

Comment:
This function seams to violate the single responsibility principle.

Reply:
How is that?

It is a protected method reuired by the above `getText()` method.
Should not a query containing an address have a method to convert that address to a oneline text query, so that the query can be used also in other providers?

Reply:
This is only meant as a default value for the text of the query, and will not be called if the user calls `GoecodeQuery::withText` him/her self to set the text portion of the query.

(Btw. "seams" = sömmar,  "seems"= verkar)